File: src/main/java/uk/co/threebugs/mochiwhattotrade3/setupgroups/SetupGroupsRepository.java
package uk.co.threebugs.mochiwhattotrade3.setupgroups;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.repository.query.QueryByExampleExecutor;

import java.util.Optional;

public interface SetupGroupsRepository extends JpaRepository<SetupGroups, Integer>, QueryByExampleExecutor<SetupGroups> {

    Optional<SetupGroups> findByScriptsDirectory(String scriptsDirectory);

    Optional<SetupGroups> findByName(String name);

}

File: src/main/kotlin/uk/co/threebugs/mochiwhattotrade3/setupgroups/SetupGroupsRepository.kt
package uk.co.threebugs.mochiwhattotrade3.setupgroups

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.repository.query.QueryByExampleExecutor
import java.util.*

interface SetupGroupsRepository : JpaRepository<SetupGroups, Int>, QueryByExampleExecutor<SetupGroups> {
    fun findByScriptsDirectory(scriptsDirectory: String): Optional<SetupGroups>
    fun findByName(name: String): Optional<SetupGroups>
}
